### PR TITLE
11 namelist description

### LINKED ISSUE #11: namelist description
We need a full description of all the namelist files needed by CABLE:

 - cable.nml
 - pft_params.nml
 - cable_soilparms.nml

Some of that documentation can be found in JULES documentation in .rst format. Some online converter tools exist, for example, https://cloudconvert.com/rst-to-md 

This is part of #6

### PR DESCRIPTION
**Pull request by tammasloughran**
# CABLE

Thank you for submitting a pull request to the CABLE Project.

## Description

Please include a brief summary of the change and list the issues that are fixed.
Please also include relevant motivation and context.

You can link issues by using a supported keyword in the pull request's description or in a commit message:

Partially fixes #11

## Type of change

Please delete options that are not relevant.

- [x] New or updated documentation

## Checklist

- [x] The new content is accessible and located in the appropriate section.
- [x] I have checked that links are valid and point to the intended content.
- [x] I have checked my code/text and corrected any misspellings

Please add a reviewer when ready for review.

This pull request adds a table for pft_params.nml to the mkdocs user guide in `documentation/docs/user_guide/inputs/pft_params_nml.md`.

The table at the has variables that are either not used or I couldm't find much information on. I took this information from:

http://jules-lsm.github.io/latest/namelists/cable_pftparm.nml.html
and 
https://trac.nci.org.au/trac/cable/wiki/CableUserGuide/Appendices

Please advise if these should be removed. I would also like someone to double check the information in the table, to see if there is anything in there that doesn't look right.

The changes to `documentation/docs/user_guide/inputs/cable_nml.md` was a quick table I made from the existing namelist on that page, before I spoke to Ramzi. The only information that I had added was for `icycle>10` as per Gab's request. The rest of the table can be overwritten at any point in the future by Ramzi.

### CODE REVIEW

**Review by rkutteh — COMMENTED** [review]
With regard to the cable_nml.md file, the table that you made form the existing namelist example for offline applications is very useful but I think it should be added to the existing example rather than replace it. As per previous discussion with @JhanSrbinovsky  and @ccarouge , we think that including input files as MD code blocks is advantageous as clicking in this case the icon in the upper right hand corner conveniently copies the entire block, which a user can then paste in their input file. So in short if you could retain the existing code and add your table to it that would be best.

A similar comment applies to pft_params_nml.md . With regard to the variables at the end which could be obsolete I think @JhanSrbinovsky and @ccarouge could shed more light on this issue as I am liable to give wrong advice if I just grep through the code (I noticed some obsolete stuff still appears inside). @ccarouge It would be useful (especially for developers) if we could create a table (say under developer_guide) that lists all quantities that have been dropped in the latest code version...I willl create a separate issue for this though.

**Comment by tammasloughran** [comment]
Thanks, Ramzi. I didn't know about that. I put the full namelist files back in the user guide.

On another topic, I have a minor criticism of Marterial for MkDocs. The web page that is generated does not make efficient use of space when displaying a large table, like the one here. The content of the page is always placed in the middle column _at a set maximum width_ no matter how wide the browser window is or how magnified the page is. This means that the table will always need to be scrolled to the left or right to view all of a single entry.
image
It would be good to be able to tweak the theme so that the content width resizes a bit more. After tinkering with CSS in the browser I found that increasing the `max-width` rule of `.md-grid` to something like 80rem makes the table much easier to view.
```CSS
.md-grid {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
}
```
@ccarouge , can we please put this in a customisation file as described by https://squidfunk.github.io/mkdocs-material/customization/ sometime?

**Comment by ccarouge** [comment]
> @ccarouge , can we please put this in a customisation file as described by https://squidfunk.github.io/mkdocs-material/customization/ sometime?

Set as an issue.

**Comment by tammasloughran** [comment]
> > @ccarouge , can we please put this in a customisation file as described by https://squidfunk.github.io/mkdocs-material/customization/ sometime?
> 
> Set as an issue #114

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 1]
```diff
@@ -84,6 +84,76 @@ applications. The following are annotated examples of cable.nml:
...
+| cable_user%FWSOIL_SWITCH     | character | 'standard' 'non-linear' 'Lai and Ktaul 2000' | 'standard'                                                                   |                                                                                             |
+|                              |  |                                          | 1. standard                                                                  |                                                                                                          |
+|                              |      |                                          | 2. non-linear extrapolation                                                  |                                                                                                          |
+|                              |      |                                          | 3. Lai and Ktaul 2000                                                        |                                                                                                          |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:146]: We don't need these lines. The available values are in the 3rd column

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 2]
```diff
@@ -84,6 +84,76 @@ applications. The following are annotated examples of cable.nml:
...
+| redistrb                     | logical |                                          | .FALSE.                                                                      | Turn on/off the hydraulic redistribution                                                                 |
+| wiltParam                    | real |                                          | 0.5                                                                          |                                                                                                          |
+| satuParam                    | real |                                          | 0.8                                                                          |                                                                                                          |
+| cable_user%FWSOIL_SWITCH     | character | 'standard' 'non-linear' 'Lai and Ktaul 2000' | 'standard'                                                                   |                                                                                             |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:143]:
```suggestion
| cable_user%FWSOIL_SWITCH     | character | 'standard' 'non-linear extrapolation' 'Lai and Ktaul 2000' | 'standard'                                                                   |                                                                                             |
```

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 3]
```diff
@@ -84,6 +84,76 @@ applications. The following are annotated examples of cable.nml:
...
+| cable_user%LEAF_RESPIRATION  |      |                                          | 'ON '                                                                        |                                                                                                          |
+| cable_user%RUN_DIAG_LEVEL    |      |                                          | 'BASIC'                                                                      | choices are:                                                                                             |
+|                              |      |                                          | 1. BASIC                                                                     |                                                                                                          |
+|                              |      |                                          | 1. NONE                                                                      |                                                                                                          |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:151]:
```suggestion
| cable_user%RUN_DIAG_LEVEL    |      |    'BASIC'  'NONE'                                    | 'BASIC'                                                                      | ```

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 4]
```diff
@@ -84,6 +84,76 @@ applications. The following are annotated examples of cable.nml:
...
+| filename%restart_out         | character |                                          | './restart_out.nc'                                                           | Output restart file                                                                                       |
+| filename%type                | character |                                          | '/projects/access/CABLE-AUX/offline/gridinfo_CSIRO_1x1.nc'                   | Gridinfo file                                                                                      |
+| filename%veg                 | character |                                          | '/projects/access/CABLE-AUX/core/biogeophys/veg_params_vbeta1.txt'           | Vegetation parameters file (obsolete, vegetation parameters are now in the pft_params.nml file)     |
+| filename%soil                | character |                                          | '/projects/access/CABLE-AUX/core/biogeophys/def_soil_params.txt'             | Soil parameters file ((obsolete, vegetation parameters are n:wow in the cable_soilparm.nml file)    |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:96]: These two options are deprecated and have no effect anymore. They are still everywhere in the namelist files that are publicly available. Should we add a column for "deprecated"?

**Review by ccarouge — CHANGES_REQUESTED** [review]

File: documentation/docs/user_guide/inputs/cable_nml.md (hunk 4 above)
Comment (tammasloughran) [reply documentation/docs/user_guide/inputs/cable_nml.md:96]: Now that Ramzi has created a deprecated features section, wouldn't this be a good candidate to put in there?

**Comment by JhanSrbinovsky** [comment]
Sounds right. They should be totally removed one rainy day
Jhan

From: Tammas Loughran ***@***.***>
Date: Thursday, 16 March 2023 at 2:34 pm
To: CABLE-LSM/CABLE ***@***.***>
Cc: Srbinovsky, Jhan (Environment, Aspendale) ***@***.***>, Mention ***@***.***>
Subject: Re: [CABLE-LSM/CABLE] 11 namelist description (PR #109)

@tammasloughran commented on this pull request.

________________________________

In documentation/docs/user_guide/inputs/cable_nml.md<https://github.com/CABLE-LSM/CABLE/pull/109#discussion_r1138066784>:

> +| filename%veg                 | character |                                          | '/projects/access/CABLE-AUX/core/biogeophys/veg_params_vbeta1.txt'           | Vegetation parameters file (obsolete, vegetation parameters are now in the pft_params.nml file)     |

+| filename%soil                | character |                                          | '/projects/access/CABLE-AUX/core/biogeophys/def_soil_params.txt'             | Soil parameters file ((obsolete, vegetation parameters are n:wow in the cable_soilparm.nml file)    |

Now that Ramzi has created a deprecated features section, wouldn't this be a good candidate to put in there?

File: documentation/docs/user_guide/inputs/cable_nml.md (hunk 4 above)
Comment (ccarouge) [reply documentation/docs/user_guide/inputs/cable_nml.md:96]: Sure, but then we need to:

- add a note at the top of the page to check the deprecated feature page if people have a file with options that are not listed here. [In an admonition](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage).
- remove these entries from the annotated examples as well.

**Comment by JhanSrbinovsky** [comment]
Perhaps just remove mention of them, they are superseded by namelists anyway offline and JAC (NOT the updated ESM1.5, CM2 though as this would involve hacking into the UM versions foreach which we don't want to do).

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 5]
```diff
@@ -68,22 +72,82 @@ applications. The following are annotated examples of cable.nml:
...
+| filename%type                | character |                                     | '/projects/access/CABLE-AUX/offline/gridinfo_CSIRO_1x1.nc'                   | Gridinfo file                                                                                            |
+| filename%veg                 | character |                                     | '/projects/access/CABLE-AUX/core/biogeophys/veg_params_vbeta1.txt'           | Vegetation parameters file (obsolete, vegetation parameters are now in the pft_params.nml file)          |
+| filename%soil                | character |                                     | '/projects/access/CABLE-AUX/core/biogeophys/def_soil_params.txt'             | Soil parameters file ((obsolete, vegetation parameters are n:wow in the cable_soilparm.nml file)         |
+| vegparmnew                   | logical   | .TURE. .FALSE.                      | .TRUE.                                                                       | Use new format for vegetation parameter files when true                                                  |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:96]:
```suggestion
| vegparmnew                   | logical   | .TRUE. .FALSE.                      | .TRUE.                                                                       | Use new format for vegetation parameter files when true                                                  |
```
The typo is in ALL the occurrences in that column! Quite a few to correct.

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 6]
```diff
@@ -68,22 +72,82 @@ applications. The following are annotated examples of cable.nml:
...
+| cable_user%CONSISTENCY_CHECK | logical   | .TURE. .FALSE.                      | .TRUE.                                                                       | TRUE outputs combined fluxes at each timestep for comparison to a control run                            |
+| cable_user%CASA_DUMP_READ    | logical   | .TURE. .FALSE.                      | .FALSE.                                                                      | TRUE reads CASA forcing from netcdf format                                                               |
+| cable_user%CASA_DUMP_WRITE   | logical   | .TURE. .FALSE.                      | .FALSE.                                                                      | TRUE outputs CASA forcing in netcdf format                                                               |
+| cable_user%SSNOW_POTEV       | character |                                     | 'HDM'                                                                        | Humidity Deficit Method                                                                                  |
+
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:150]:
```suggestion
| cable_user%SSNOW_POTEV       | character(len=3) |   'HDM' 'P-M'                         | 'HDM'                                                                        | "Humidity Deficit Method" or  "Penman-Monteith"                                                                    |

```

**Review by ccarouge — CHANGES_REQUESTED** [review]
A few more changes sorry.

I'll also add information on the length of the character variables. But I'll do that through a direct commit instead of a comment.

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 7]
```diff
@@ -68,22 +72,82 @@ applications. The following are annotated examples of cable.nml:
...
+| filename%log                 | character |                                     | 'log_cable.txt'                                                              | Log file                                                                                                 |
+| filename%restart_in          | character |                                     | ' '                                                                          | Input restart file                                                                                       |
+| filename%restart_out         | character |                                     | './restart_out.nc'                                                           | Output restart file                                                                                      |
+| filename%type                | character |                                     | '/projects/access/CABLE-AUX/offline/gridinfo_CSIRO_1x1.nc'                   | Gridinfo file                                                                                            |
```
[the remark this answers is not in the document]
Comment (tammasloughran) [reply documentation/docs/user_guide/inputs/cable_nml.md]: Sure. The fact it's called grid is a bit confusing. I just assumed it had information on the grid structure, but it looks like it has other stuff about soil and vegetation. Could elaborate... "about soil and vegetation". For some variables like soil moisture and temperature I'm wondering if they really are static?

**Comment by tammasloughran** [comment]
I have nothing more to add. Can I suggest a merge?

File: documentation/docs/user_guide/inputs/cable_nml.md [hunk 8]
```diff
@@ -30,7 +30,7 @@ applications. The following are annotated examples of cable.nml:
    soilparmnew = .TRUE.  ! using new format when true
    spinup = .TRUE.  ! do we spin up the model?
    delsoilM = 0.001   ! allowed variation in soil moisture for spin up
-   delsoilT = 0.01    ! allowed variation in soil temperature for spin up
+   delsoilT = 0.01    ! allowed variation in soil temperaTRUE for spin up
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/cable_nml.md:33]:
```suggestion
   delsoilT = 0.01    ! allowed variation in soil temperature for spin up
```

**Review by ccarouge — APPROVED** [review]
Just one tiny modification required. I'll approve it in and then @tammasloughran you can merge it in. Great job!

**Pull request merged**